geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/ExpressionParser.scala

```diff
@@ -60,7 +60,7 @@ private [transforms] class ExpressionParser extends BasicParser {
   }
 
   private def column: Rule1[Expression] = rule("$col") {
-    "$" ~ int ~~> { i => if (i == 0) { WholeRecord } else { Column(i) } }
+    "$" ~ int ~~> { i => Column(i) }
```

Comment:
Is $0 still the whole record?  Or does this change that?

Reply:
still the whole record, just removed the case object as it's not really needed

Reply:
Above in the documentation, $0 is the id-field?

Reply:
for shp converters, $0 is the feature ID, for delimited text it's the whole record

Reply:
that was the impetus to change it actually, since whole record is specific to delimited text

Reply:
Hmm....  I'm slightly nervous about $0 meaning something different in the two different contexts.

We recommend md5sum($0) as a way to roll a unique ID, and that'd be different between the same data in Shapefile and GeoJson format...

Reply:
my reasoning was that $0 will still be unique, since it's the feature ID. it already means a few different things though - for example in json it's just the result of the json-path expression